app/rtc_compare.py

```diff
@@ -0,0 +1,408 @@
+    ----------
+    hdf_obj_1: h5py.Group
+        HDF5 object to retrieve the dataset and the attribute list
+    key_in: str
+        path in the HDF5 object
+    list_dataset_so_far: list
+        list of the dataset keys that have found so far
+    list_attrs_so_far: list
+        list of the attribute path/keys that have found so far
+
+    Return:
+    -------
+    _ : 0 if the two HDF are identical; 1 otherwise
+
+    '''
+
+    # default values for the lists
+    if list_dataset_so_far is None:
+        list_dataset_so_far = []
+    if list_attrs_so_far is None:
+        list_attrs_so_far = []
+
+    if isinstance(hdf_obj_1[key_in], h5py.Group):
+        # Append the attributes keys if there are any
+        for key_attr_1 in hdf_obj_1[key_in].attrs:
+            list_attrs_so_far.append('\n'.join([key_in, key_attr_1]))
+
+        for key_1, _ in hdf_obj_1[key_in].items():
+            get_list_dataset_attrs_keys(hdf_obj_1, f'{key_in}/{key_1}',
+                                        list_dataset_so_far,
+                                        list_attrs_so_far)
+
+    else:
+        list_dataset_so_far.append(key_in)
+        for key_attr_1 in hdf_obj_1[key_in].attrs:
+            # Append the attributes keys if there are any
+            list_attrs_so_far.append('\n'.join([key_in, key_attr_1]))
+    return list_dataset_so_far, list_attrs_so_far
+
+
+def compare_dataset_attr(hdf5_obj_1, hdf5_obj_2, str_key, is_attr=False):
+    '''
+    Compare the dataset or attribute defined by `str_key`
+    NOTE: For attributes, the path and the key are separated by newline character ('\n')
+
+    Parameters:
+    -----------
+    hdf5_obj_1: h5py.Group
+        The 1st HDF5 object to compare
+    hdf5_obj_2: h5py.Group
+        The 2nd HDF5 object to compare
+    str_key: str
+        Key to the dataset or attribute
+    is_attr: bool
+        Designate if `str_key` is for dataset or attribute
+
+    Return:
+    -------
+    _: True when the dataset are identical; False otherwise
+    '''
+
+    # Prepare to comapre the data in the HDF objects
+    if is_attr:
+        # str_key is for attribute
+        path_attr, key_attr = str_key.split('\n')
+        val_1 = hdf5_obj_1[path_attr].attrs[key_attr]
+        val_2 = hdf5_obj_2[path_attr].attrs[key_attr]
+        # Force the datatype of val_1 and _2 to make use of numpy features
+        if not isinstance(val_1,np.ndarray):
+            val_1 = np.array(val_1)
+
+        if not isinstance(val_2,np.ndarray):
+            val_2 = np.array(val_2)
+
+    else:
+        # str_key is for dataset
+        val_1 = np.array(hdf5_obj_1[str_key])
+        val_2 = np.array(hdf5_obj_2[str_key])
+
+    shape_val_1 = val_1.shape
+    shape_val_2 = val_2.shape
+
+    if shape_val_1 != shape_val_2:
+        # Dataset or attribute shape does not match
+        print(f'    - Data shapes do not match. {shape_val_1} vs. {shape_val_2}')
+        return False
+
+    if len(shape_val_1)==0 and len(shape_val_2)==0:
```

Comment:
```suggestion
    if len(shape_val_1)==0:
```

You already checked that `shape_val_1` is equal to `shape_val_2`.

Reply:
>Well, but you'll are comparing all elements of the HDF5 dataset. Soon or later that element (that was referenced by another one) will be checked as well.

Now we just the extract the string data in the list, we can convert it to `np.ndarray` in a safer way. No need to differentiate the comparison strategy anymore. I've made the modification on the code.